### PR TITLE
feat: Add support for tab in autocompletion

### LINKED ISSUE #20448: [Feature]: Add support for Tabs in autocompletion
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Today, only the "Enter" key is used to select an autocompletion hint. Most editors today allow both Tab and Enter keys to select hints.

### Why should this be worked on?

Some users are already accustomed to using the Tab key, not supporting this will break their usual flow

### PR DESCRIPTION
**Pull request by ravikp7**
## Description
This adds the ability to select an option from the autocomplete suggestions list using the `Tab` key while writing code. Earlier, users could use only the `Enter` key to do so. All code editors support `Tab` for autocomplete.

Fixes #20448 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Cypress

### Test Plan
[Test Plan](https://github.com/appsmithorg/TestSmith/issues/2377)

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by Rishabh-Rathod — DISMISSED** [review]
LGTM

**Review by Aishwarya-U-R — DISMISSED** [review]

**Comment by Aishwarya-U-R** [comment]
@ravikp7 - Cypress looks Ok, lets check its passing in CI too.

**Comment by shwetha-ramesh** [comment]
@ravikp7 Tested on this [DP](https://appsmith-bgiqoje23-get-appsmith.vercel.app/). Working as expected. Moving to done.

File: app/client/package.json [hunk 1]
```diff
@@ -268,6 +268,7 @@
     "cypress-file-upload": "^4.1.1",
     "cypress-image-snapshot": "^4.0.1",
     "cypress-multi-reporters": "^1.2.4",
+    "cypress-plugin-tab": "^1.0.5",
```
Comment (Aishwarya-U-R) [inline app/client/package.json:271]: This should be added into package.json under cypress folder also. The references from that package is what is used by cypress tests.

Comment (Aishwarya-U-R) [reply app/client/package.json:271]: Want me to approve PR?

**Pull request merged**